### PR TITLE
Added jax.numpy prod to jax frontend

### PR DESCRIPTION
**Pull request by leacornelio**

### CODE REVIEW

**Review by kurshakuz — CHANGES_REQUESTED** [review]
Hello @leacornelio ! Can you please resolve the conflict and link the associated issue? Thanks 🙂

**Comment by leacornelio** [comment]
> Hello @leacornelio ! Can you please resolve the conflict and link the associated issue? Thanks 🙂

Hi. Conflicts are now resolved. Thanks!

**Comment by leacornelio** [comment]
Closing #10309

File: ivy/functional/frontends/jax/numpy/mathematical_functions.py [hunk 1]
```diff
@@ -425,6 +425,11 @@ def floor_divide(x1, x2, /, out=None):
 
 
 @to_ivy_arrays_and_back
+def prod(x, /, out=None):
+    return ivy.prod(x, out=out)
+
+
 def inner(a, b):
```
Comment (kurshakuz) [inline ivy/functional/frontends/jax/numpy/mathematical_functions.py:432]: Please return the initial `@to_ivy_arrays_and_back` decorator back to the `inner` function. You don't need to change it
```suggestion

@to_ivy_arrays_and_back
def inner(a, b):
```

**Review by kurshakuz — CHANGES_REQUESTED** [review]
Hello @leacornelio !


Your implementation is failing due to following issue:

```
E       AssertionError: the return with a jax backend produced data type of float32, while the return with a tensorflow backend returned a data type of float64.
E       Falsifying example: test_jax_numpy_prod(
E           dtype_and_x=(['float32'], [array(-1., dtype=float32)]),
E           test_flags=num_positional_args=1. with_out=False. inplace=False. native_arrays=[False]. as_variable=[False]. ,
E           fn_tree='ivy.functional.frontends.jax.numpy.prod',
E           on_device='cpu',
E           frontend='jax',
E       )
```

Please resolve it and let me know once you are ready!

File: ivy/functional/frontends/jax/numpy/mathematical_functions.py (hunk 1 above)
Comment (leacornelio) [reply ivy/functional/frontends/jax/numpy/mathematical_functions.py]: Sorry I might have accidentally removed the decorator. Brought it back and set the data type for the jax numpy prod function.

**Review by kurshakuz — CHANGES_REQUESTED** [review]
Your implementation is still failing local tests, can you please take a look and fix it? Thanks!

```
E           hypothesis.errors.InvalidArgument: Cannot sample from a length-zero sequence.
E           
E           You can reproduce this example by temporarily adding @reproduce_failure('6.55.0', b'AA==') as a decorator on your test case
```

**Comment by leacornelio** [comment]
Apologies @kurshakuz I've been trying to do frontend tests but I'm getting stuck with this error
`AttributeError: 'NoneType' object has no attribute 'valid_dtypes'`. Do you know what's wrong?

**Pull request closed without merging**